hazelcast/src/main/java/com/hazelcast/datalink/DataLinkService.java

```diff
@@ -20,34 +20,89 @@
+ * <p>
+ * {@link DataLink}s defined in the configuration are created at startup.
+ * {@link DataLink}s added to the configuration dynamically are created via
+ * {@link #createConfigDataLink(DataLinkConfig)}.
+ * {@link DataLink}s created via SQL are added via {@link #createSqlDataLink(String, String, Map)},
+ * these can be removed via {@link #removeDataLink(String)}.
+ * <p>
+ * When a new config is added via {@link #createConfigDataLink(DataLinkConfig)} is created and SQL DataLink already
+ * exists the existing DataLink is replaced and closed.
  *
- * @since 5.2
+ * @since 5.3
  */
 @Beta
 public interface DataLinkService extends AutoCloseable {
 
     /**
-     * Tests data link configuration.
+     * Creates a new DataLink with the given config
+     * <p>
+     * Such DataLink is considered immutable and can't be updated.
+     * The DataLink is closed when {@link #close()} is called.
+     *
+     * @param config the configuration of the DataLink
+     */
+    void createConfigDataLink(DataLinkConfig config);
+
+    /**
+     * Creates a new DataLink with the given parameters.
      *
-     * @param config name of the data link factory
-     * @return {@code true} if test was successful
-     * @throws Exception if the test operation fails
-     * @since 5.3
+     * @param name    name of the DataLink
+     * @param type    type of the DataLink
+     * @param options options configuring the DataLink
      */
-    boolean testConnection(DataLinkConfig config) throws Exception;
+    void createSqlDataLink(String name, String type, Map<String, String> options);
 
     /**
-     * Returns data link factory with given name.
+     * Replaces an existing DataLink with a new one with given parameters.
+     * <p>
+     * The old DataLink is closed.
      *
-     * @param name name of the data link factory
-     * @param <DL> type of the data link
-     * @return instance of the factory
-     * @throws HazelcastException if the factory with given name is not found or misconfigured*
+     * @param name    name of the DataLink
+     * @param type    type of the DataLink
+     * @param options options configuring the DataLink
      */
-    <DL> DataLinkFactory<DL> getDataLinkFactory(String name);
+    void replaceSqlDataLink(String name, String type, Map<String, String> options);
 
+    /**
+     * Returns DataLink with given name.
+     *
+     * @param name name of the DataLink
+     * @param <T>  type of the DataLink
+     * @return instance of the DataLink
+     * @throws HazelcastException if the DataLink with given name is not found or misconfigured*
+     */
+    <T extends DataLink> T getDataLink(String name);
+
+    /**
+     * Returns DataLink with given name.
+     * <p>
+     * Type checked against the provided clazz parameter.
+     *
+     * @param name name of the DataLink
+     * @param clazz type of the DataLink
+     */
+    <T extends DataLink> T getDataLink(String name, Class<T> clazz);
```

Comment:
What about `isDataLinkExists()`? It would be useful for SQL to have check method which doesn't throw.

Reply:
25d79f6bb5be89fe249480510da3288ad7f6986f